lesson-2.js

```diff
@@ -0,0 +1,139 @@
+'use strict'
+//Task 1
+// var a = 1, b = 1, c, d;
+// c = ++a; alert(c);           // 2, тк сперва выполнится инкремент и а увеличится на 1
+// d = b++; alert(d);           // 1, тк сперва идет вывод, а затем инкремент
+// c = (2+ ++a); alert(c);      // 5, в а уже лежит 2, мы инкрементом увеличиваем еще на 1 и просто прибавляем 2
+// d = (2+ b++); alert(d);      // 4, аналогично предыдущему, но b увеличится на 1 после вывода
+// alert(a);                    // 3, после всех операций в а осталась 3
+// alert(b);                    // 3, после вывода сработал инкремент, b увеличилось на 1
+
+//Task 2
+// var a = 2;
+// var x = 1 + (a *= 2); 
+//x = 5, тк (a *= 2) = 4
+
+//Task 3
+var a = 10;
+var b = 15;
+function compare() {
+  if (a > 0 && b > 0) {
+    alert(a - b);
+  }
+  else if (a < 0 && b < 0) {
+    alert(a * b);
+  }
+  else if ( (a < 0 && b > 0) || (a > 0 && b < 0)) {
```

Comment:
Здесь нет смысла в дополнительном условии, т.к. по сути нас интересуют "все остальные" случаи, которые прекрасно покрываются обычным else
```suggestion
  else {
```